### ISSUE TITLE
AI Quirk Zone

### ISSUE THREAD

**Issue by Gidoza** [LoneGazebo/Community-Patch-DLL#2171]
Not sure the best way to do it - but Gazebo suggested an AI Quirks thread in GitHub, so I thought I'd take up the initiative.  Just post your observations here, with evidence if you have it.  Perhaps we can make a numbered list and keep track so Gazebo can check things and definitively cross things off.  Well anyways, that's the idea.  =)

**Comment by Gidoza** [comment]
Issue #1 - Ranged horse unit is pinned between two of my own units and is guaranteed to die - it *COULD* shoot and then move away, but the AI always just moves a single tile without firing.  It could at least be pragmatic about things and try to kill something before moving.

**Comment by ilteroi** [comment]
that one was easy - should be better now :)

**Comment by LoneGazebo** [comment]
@ilteroi, I've got one for you. :) I've noticed that embarked armies (i.e. during a naval invasion) are lacking proper naval unit support. I think it would make sense to have embarked units on an operation count as tactical necessities for naval units to move towards and defend.

**Comment by Gidoza** [comment]
Thanks for that fix, ilteroi!

I'll support Gazebo as well on his comment, though perhaps I can take it a bit further (though I don't know how much further YOU can take it).  The AI is so insistent on crossing the ocean with its embarked troops that it will even attempt to by-pass an entire fleet of ships that is just sitting there waiting on the other side.  Some kind of script that makes the AI think, "Hmmmm...maybe I shouldn't bother till later," could be useful.  :)

**Comment by LoneGazebo** [comment]
I feel like a possible avenue for this would be naval sentry functionality. Right now, the AI is absolutely spot-on with land defense and sentry functionality. At sea, though, it flounders (pun intended). Any chance we can copy the land patrol function, make a naval one, and adjust it so that naval units that are not in an army are actively looking for embarked units to sit on or chokepoints/harbors to defend?

**Comment by Iamblichos** [comment]
I think that would be worthwhile. It's almost impossible to do a land sneak attack because of sentries, but naval sneak attacks are relatively easy.

**Comment by LoneGazebo** [comment]
I'm working on naval sentries now. :)

**Comment by Gidoza** [comment]
Found another oddity when playing a game yesterday.

Enemy Spearman in enemy city that is almost dead - the opponent could have just opted to do nothing, heal 20 hit points, and make my assault on the city that much more difficult for another turn (or indefinitely, if I can't do enough damage).  But instead of doing that - my opponent attacks my Spearman with his Spearman; his Spearman is killed, his city is left without a unit in it and so much easier to attack, and moreover my Authority Policy heals my unit for 20 hit points, when if the city had fired first, perhaps my Spearman would have died.  Everything is wrong with this situation.  :P

**Comment by ilteroi** [comment]
@g1: should be possible, yes. it's just for historical reasons that ships have different moves. definitely room for improvement there.

@g2: did you save the autosave? cause that shouldn't be happening, except if your attack was the only one around and the AI unit had a decent chance of killing it.

**Comment by LoneGazebo** [comment]
@ilteroi I've got naval sentries working. Now we just need to figure out naval protection of embarked units.

**Comment by Gidoza** [comment]
Sorry, didn't keep the auto save, but I'll watch for more - but regarding your two notes.

1.  My attacking unit wasn't the only one around (many Archers hitting the city as well).

2.  My Spearman wasn't doing great in hit points, but hardly low enough to ever be a sure kill.

3.  Furthermore, this did not happen merely once, but 5 different times.

**Comment by isaac331** [comment]
This seems belong here: friendly AI (DoF) keep making worthless trade offer for my 2nd city (when I only had 2) since early game. More like an insult than a reasonable offer, which is ironic because it comes from a supposedly 'friend'. 

He regularly make casual bid for my booming city yet when I try to bid one of his 5 junk cities, most are 'impossible' and one is valued at 16K while he values my city at only 1300 (which he offered about 35gpt for). Seems like double standards.

I think the AI needs some work on this aspect. Maybe make the AI only bid for cities if a player has more than 4 cities and his empire is not happy (sell city = quick happiness), preferably with fair trade value. Or maybe just stop the AI from actively bidding for cities.

**Comment by mattalat** [comment]
Perhaps I should open a separate issue for this, or try to reproduce the quirk in more games, but I've come across a problem in which the AI incessantly requests that I renew an old trade deal. Past deals of charity and good will, like free luxury resources to ailing AI civs, are now requested of me by practically every civ on every turn. One civ, for instance, keeps requesting that I give them 10 gold per turn like I did back in the early stages of the game, despite the fact that they are now making 200+ gold per turn. 

No matter how many times I refuse, the AI continually requests I renew these deals. Has anyone else experienced this?

**Comment by LoneGazebo** [comment]
Continually request the same deal even after you deny?

**Comment by mattalat** [comment]
Yes. The requested deals seem to specifically be renewals for deals I had made with the civs in the past.

**Comment by LoneGazebo** [comment]
But it is the same deal over and over?

**Comment by LoneGazebo** [comment]
May be fixed in hotfix.
